src/base/solvers/solveCobraMILP.m

```diff
@@ -432,8 +436,8 @@
         end
         
         % Get results
-        x = Result.x;
-        f = osense*Result.objval;
+        try x = Result.x; catch x=zeros(size(cplexlp.Model.ub)); end %return zeros if infeasible
+        try f = osense*Result.objval; catch f=nan; end %return Nan if infeasible
```

Comment:
x should be empty if its infeasible. 
There are enough instances, where people do not check for the status of the solution and would use the zero vector.

Reply:
The problem with ibm_cplex is that for infeasible problems, there is no field in Results called x.
Whenever the code arrive to this line, this produces an error. The test to confirm if the solution is optimal or not is done in the following lines of code.
The change in code was done to return a solution similar to gplk with infeasible problems.

Reply:
I would simply move the lines into the first if block (i.e. after `if (stat == 101 || stat == 102 || stat == 1)`).
If the solution is not optimal, don't return something that looks like a solution. and by default (see the lines 191-194), `x` and `f` are set to `[]`.